fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -1288,6 +1289,45 @@ public void isAbsentBuildable() throws Exception {
         }
     }
 
+    @Test
+    public void getIndexStates() throws Exception {
+        final String indexName = "MySimpleRecord$str_value_indexed";
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            final Index index = recordStore.getRecordMetaData().getIndex(indexName);
+            assertThat(recordStore.getAllIndexStates(), hasEntry(index, IndexState.READABLE));
+            recordStore.markIndexWriteOnly(indexName).get();
+            assertThat(recordStore.getAllIndexStates(), hasEntry(index, IndexState.WRITE_ONLY));
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            final Index index = recordStore.getRecordMetaData().getIndex(indexName);
+            assertThat(recordStore.getAllIndexStates(), hasEntry(index, IndexState.WRITE_ONLY));
+            recordStore.markIndexDisabled(indexName).get();
+            assertThat(recordStore.getAllIndexStates(), hasEntry(index, IndexState.DISABLED));
+            // does not commit
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            final Index index = recordStore.getRecordMetaData().getIndex(indexName);
+            assertThat(recordStore.getAllIndexStates(), hasEntry(index, IndexState.WRITE_ONLY));
+            recordStore.saveRecord(TestRecords1Proto.MySimpleRecord.newBuilder()
```

Comment:
Should this be a `MyOtherRecord` instead of a `MySimpleRecord`? I think (or I hope) that adding the record and changing one of the state of one of the indexes defined on that record type should already conflict. (Or, in other words, I would expect this test to pass even if line 1317 were commented out.)